packages/viewer/src/lib/types/__tests__/sceneViewSummary.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { BoundingBox, Vector3 } from '@vertexvis/geometry';
+
+import { makePerspectiveFrame } from '../../../testing/fixtures';
+import { Frame, SceneViewSummary } from '..';
+
+describe('SceneViewSummary', () => {
+  describe('SceneViewSummary.copySummaryIfInvalid', () => {
+    const baseFrame = makePerspectiveFrame();
+
+    it('copies the summary of the previous frame if the current summary is invalid', () => {
+      const previousValid = copyWithValidSummary(baseFrame, 1);
+      const previousInvalid = copyWithInvalidSummary(baseFrame);
+      const current = copyWithInvalidSummary(baseFrame);
+
+      expect(
+        SceneViewSummary.copySummaryIfInvalid(current, previousValid).scene
+          .sceneViewSummary
+      ).toMatchObject(previousValid.scene.sceneViewSummary);
+      expect(
+        SceneViewSummary.copySummaryIfInvalid(current, previousInvalid).scene
+          .sceneViewSummary
+      ).toMatchObject(current.scene.sceneViewSummary);
+    });
+
+    it('uses the summary of the current frame if it is valid', () => {
+      const previousValid = copyWithValidSummary(baseFrame, 1);
+      const previousInvalid = copyWithInvalidSummary(baseFrame);
+      const current = copyWithValidSummary(baseFrame, 2);
+
+      expect(
+        SceneViewSummary.copySummaryIfInvalid(current, previousValid).scene
+          .sceneViewSummary
+      ).toMatchObject(current.scene.sceneViewSummary);
+      expect(
+        SceneViewSummary.copySummaryIfInvalid(current, previousInvalid).scene
+          .sceneViewSummary
+      ).toMatchObject(current.scene.sceneViewSummary);
+    });
+  });
+});
+
+function copyWithSummary(
+  frame: Frame,
+  summary?: SceneViewSummary.SceneViewSummary
+): Frame {
+  return frame.copy({
+    scene: frame.scene.copy({
+      sceneViewSummary: summary,
+    }),
+  });
+}
+
+function copyWithValidSummary(frame: Frame, count: number): Frame {
+  return copyWithSummary(frame, {
+    selectedVisibleSummary: {
+      count,
+      boundingBox: BoundingBox.create(Vector3.create(), Vector3.right()),
+    },
+    visibleSummary: {
+      count,
+      boundingBox: BoundingBox.create(Vector3.create(), Vector3.right()),
+    },
+  });
+}
+
+function copyWithInvalidSummary(frame: Frame): Frame {
+  return copyWithSummary(frame, {
+    selectedVisibleSummary: { count: -1 },
```

Comment:
Should this be `selectedVisibleSummary: null`?

Reply:
Not currently with the way this is set up, as the `visibleSummary` and `selectedVisibleSummary` will still always come back from FSS with a count of `-1` if it's invalid (as the `count` on the item set summary is required). We could consider making more things at the FSS level optional though to avoid this, or support undefined values in the SDK though they'd never truly be undefined